### PR TITLE
Fix amqp instrumentation

### PR DESCRIPTION
**Pull request by piochelepiotr**
### What does this PR do?

The amqp instrumentation was not getting the queue name correctly.
In some cases (when using the Get function) it was also not propagating context correctly.

This PR addresses these two issues.

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

File: .github/workflows/system-tests.yml [hunk 1]
```diff
@@ -54,6 +54,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           repository: 'DataDog/system-tests'
+          ref: 'piotr-wolski/update-node-js-test'
```
Comment (wconti27) [inline .github/workflows/system-tests.yml:57]: This will need to be removed before merge

Comment (piochelepiotr) [reply .github/workflows/system-tests.yml:57]: Yes, it goes with this PR: https://github.com/DataDog/system-tests/pull/3346

Not sure in what order I should merge them (probably tests first)

**Comment by tlhunter** [comment]
Can you add a test to ensure that the broken scenario is now fixed?

**Review by tlhunter — APPROVED** [review]

**Comment by piochelepiotr** [comment]
I merged https://github.com/DataDog/system-tests/pull/3346 & updated the system test branch to be main branch again.

**Review by tlhunter — APPROVED** [review]

**Pull request merged**